extensions/binding/org.eclipse.smarthome.binding.dmx/src/main/java/org/eclipse/smarthome/binding/dmx/internal/multiverse/Channel.java

```diff
@@ -0,0 +1,367 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.dmx.internal.multiverse;
+
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map.Entry;
+
+import org.eclipse.smarthome.binding.dmx.DmxBindingConstants.ListenerType;
+import org.eclipse.smarthome.binding.dmx.internal.DmxThingHandler;
+import org.eclipse.smarthome.binding.dmx.internal.Util;
+import org.eclipse.smarthome.binding.dmx.internal.action.BaseAction;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link Channel} extends {@link BaseChannel} with actions and values
+ * handlers.
+ *
+ * @author Jan N. Klug - Initial contribution
+ * @author Davy Vanherbergen
+ */
+public class Channel extends BaseChannel {
+    public static final int MIN_VALUE = 0;
+    public static final int MAX_VALUE = 255;
+
+    private final Logger logger = LoggerFactory.getLogger(Channel.class);
+
+    private int value = MIN_VALUE;
+    private int suspendedValue = MIN_VALUE;
+    private int lastStateValue = -1;
+
+    private boolean isSuspended;
+    private long lastStateTimestamp = 0;
+
+    private List<BaseAction> actions = new ArrayList<BaseAction>();
+    private List<BaseAction> suspendedActions = new ArrayList<BaseAction>();
+    private List<Thing> registeredThings = new ArrayList<Thing>();
+
+    private HashMap<ChannelUID, DmxThingHandler> onOffListeners = new HashMap<ChannelUID, DmxThingHandler>();
+    private HashMap<ChannelUID, DmxThingHandler> valueListeners = new HashMap<ChannelUID, DmxThingHandler>();
+    private Entry<ChannelUID, DmxThingHandler> actionListener = null;
```

Comment:
This could be stored in two variables. I chose to use the Entry to clearly show that these values belong together, they are set from the thing that adds the listener.